### PR TITLE
Refresh of the threat model documentation.

### LINKED ISSUE #254: Refresh reference/Threat model

### PR DESCRIPTION
**Pull request by jhkrug**
Will close #254 when merged.

There is one TODO(my todo). I'm wondering all of these todos mentioned in the document (Kubewarden team todos) are still valid, and hence the associated mitigations..

Thanks.

### CODE REVIEW

File: docs/security/threat-model.md [hunk 1]
```diff
@@ -1,223 +1,303 @@
...
 the webhook object in the cluster.
 
-**Mitigation**
+#### Mitigation
+
 RBAC rights are strictly controlled.
 
-**To do**
-Most of RBAC isn't within the scope of the current discussion. 
+#### To-do
```
Comment (flavio) [inline docs/security/threat-model.md:74]: This hasn't been done yet

File: docs/security/threat-model.md [hunk 2]
```diff
@@ -1,223 +1,303 @@
...
 can attempt to use MITM tooling to intercept traffic between the API server
 and admission controller webhook.
 
-**Mitigation**
+#### Mitigation
+
 Webhook mTLS authentication is used.
 
-**To do**
+#### To-do
```
Comment (flavio) [inline docs/security/threat-model.md:127]: Not done

File: docs/security/threat-model.md [hunk 3]
```diff
@@ -1,223 +1,303 @@
...
 An attacker is able to redirect traffic from the API server which is intended
 for the admission controller webhook by spoofing.
 
-**Mitigation**
+#### Mitigation
+
 Webhook mTLS authentication is used.
 
-**To do**
+#### To-do
```
Comment (flavio) [inline docs/security/threat-model.md:145]: not done

File: docs/security/threat-model.md [hunk 4]
```diff
@@ -1,223 +1,303 @@
...
 
-## Threat 11 - Attacker deploys workloads to namespaces that are  exempt from admission control
 An attacker is able to deploy workloads to Kubernetes namespaces that are exempt
 from the admission controller configuration.
 
-**Mitigation**
-RBAC rights are strictly controlled
+#### Mitigation: RBAC rights are strictly controlled
+
+#### To-do
```
Comment (flavio) [inline docs/security/threat-model.md:174]: not done

File: docs/security/threat-model.md [hunk 5]
```diff
@@ -1,223 +1,303 @@
...
+#### Mitigation
+
 All rules are reviewed and tested.
 
-**To do**
+#### To-do
+
 Introduce tests to cover this rule.
-As always, carefully review PRs changing the rules in the policies deployment.
+As always, you should review PRs changing the rules in the policies deployment.
```
Comment (flavio) [inline docs/security/threat-model.md:195]: I would drop this TODO.

On our side we have to keep up with Kubernetes releases and review the changes they introduce. When init container have been introduced we reviewed our policies to ensure they would catch them too.

On the user side, they have to review their in place rules and make sure they are covering these new corner cases.

File: docs/security/threat-model.md [hunk 6]
```diff
@@ -1,223 +1,303 @@
...
+#### Mitigation
+
+All rules should be reviewed and tested.
+
+#### To-do
 
-**To do**
 Introduce tests to cover this rule.
-As always, carefully review PRs changing the rules in the policies deployment.
+As always, you should review PRs changing the rules in the policies deployment.
```
Comment (flavio) [inline docs/security/threat-model.md:211]: I would move all these TODOs under "mitigation". There's nothing we can do in this case IMHO

File: docs/security/threat-model.md [hunk 7]
```diff
@@ -1,223 +1,303 @@
...
 
-**To do**
+#### To-do
+
 Introduce tests to cover this rule.
-Create a configuration to reject requests where the API version not
-cover by the policy by default. Kubewarden should warn policy developers to cover all the
-supported API version in their tests and reject all others.
+Create a configuration to reject requests where the API version doesn't cover by the policy by default.
+Kubewarden should warn policy developers to cover all the supported API version in their tests and reject all others.
```
Comment (flavio) [inline docs/security/threat-model.md:228]: We now have a policy that flags deprecated resources being used: https://github.com/kubewarden/deprecated-api-versions-policy

We could drop this TODO and write something like that under the mitigation:

> Rely on [deprecated-api-versions-policy](https://github.com/kubewarden/deprecated-api-versions-policy) to identify and block usage of deprecated resources.

File: docs/security/threat-model.md [hunk 8]
```diff
@@ -1,223 +1,303 @@
...
+#### Scenario
 
-## Threat 8 - Attacker carries out a MITM attack on the webhook
 An attacker on the container network, who has access to the NET_RAW capability
 can attempt to use MITM tooling to intercept traffic between the API server
 and admission controller webhook.
 
-**Mitigation**
+#### Mitigation
+
```
Comment (flavio) [inline docs/security/threat-model.md:124]: We could add a new mitigation:

> Deploy the [capabilities-psp](https://artifacthub.io/packages/kubewarden/capabilities-psp/capabilities-psp) and configure it to drop `NET_RAW` capability.

File: docs/security/threat-model.md (hunk 1 above)
Comment (jhkrug) [reply docs/security/threat-model.md:74]: Thanks.

File: docs/security/threat-model.md (hunk 2 above)
Comment (jhkrug) [reply docs/security/threat-model.md:127]: Thanks.

File: docs/security/threat-model.md (hunk 5 above)
Comment (jhkrug) [reply docs/security/threat-model.md:195]: Thanks.

File: docs/security/threat-model.md (hunk 7 above)
Comment (jhkrug) [reply docs/security/threat-model.md:228]: Done, Thanks.

File: docs/security/threat-model.md (hunk 4 above)
Comment (jhkrug) [reply docs/security/threat-model.md:174]: Thanks.

File: docs/security/threat-model.md (hunk 6 above)
Comment (jhkrug) [reply docs/security/threat-model.md:211]: Done, thanks.

File: docs/security/threat-model.md (hunk 8 above)
Comment (jhkrug) [reply docs/security/threat-model.md:124]: Done, thanks.

File: docs/security/threat-model.md (hunk 7 above)
Comment (viccuad) [reply docs/security/threat-model.md]: Unresolving.

@flavio Note that the deprecated-api-versions-policy only deals with CRs known to it.
The threat is both deprecated api versions, and new unknown versions that are misused, hence the policy only covers half of it.

For it to be covered fully, we should add an allowlist to deprecated-api-versions-policy, or create a new allowlist-api-versions-policy. This is covered in https://github.com/kubewarden/kubewarden-controller/issues/227.

File: docs/security/threat-model.md [hunk 9]
```diff
@@ -1,223 +1,288 @@
...
-All rules are reviewed and tested.
+#### Mitigation
 
-**To do**
-Introduce tests to cover this rule.
-Create a configuration to reject requests where the API version not
-cover by the policy by default. Kubewarden should warn policy developers to cover all the
-supported API version in their tests and reject all others.
+All rules should be reviewed and tested. There is a policy that tests for the use of deprecated resources. It's available from [the deprecated-api-versions-policy](https://github.com/kubewarden/deprecated-api-versions-policy).
+
```
[the remark this answers is not in the document]
Comment (viccuad) [reply docs/security/threat-model.md:214]: sounds good! @jhkrug feel free to apply the suggestion.

**Review by viccuad — APPROVED** [review]
